index.bs

```diff
@@ -692,44 +692,75 @@ When this method is invoked, the user agent MUST execute the following algorithm
+
+                1. let |Y| be a new {{ScopedCredentialDescriptor}}.
+
+                1. Set |Y|'s members according to the corresponding properties of |X|.
+
+                1. [=set/append=] |Y| to |credentialDescriptorList|.
 
-                <span class="note">Note: the procedure to determine which credentials present on |authenticator| map to 
-                    |rpId|, is platform-specific.</span>
         </dl>
-    
-    1. If |credentialList| [=list/is empty=] then [=continue=].
-
-    1. [=In parallel=], [=list/for each=] credential |C| in |credentialList|:
-        1. If <code>|C|.{{transports}}</code> [=list/is not empty=], the client SHOULD select one |transport| from
-            {{transports}}. Then, using |transport|, invoke the [=authenticatorGetAssertion=] operation on |authenticator|, with
-            |rpId|, |clientDataHash|, |credentialList|, and |clientExtensions| as parameters.
-        1. Otherwise, using local configuration knowledge of the appropriate transport to use with |authenticator|, invoke the 
-            [=authenticatorGetAssertion=] operation on |authenticator| with |rpId|, |clientDataHash|, |credentialList|, and
-            |clientExtensions| as parameters.
-    
+        <span class="note">Note: the procedures to: determine the [=authenticators=] present on a platform,
+            determine which credentials present on |authenticator| map to a given |rpId|,
+            match credentials present on |authenticator| with values of members of {{ScopedCredentialDescriptor}},
+            and obtain the properties of credentials present on an |authenticator|,
+            are platform-specific.</span>
+
+    1. If |credentialDescriptorList| [=list/is empty=] then [=continue=].
+
+    1. Let |credentialsPerTransportMap| be a new [=ordered map=] whose [=map/keys=] are values of {{Transport}}, and whose
+        [=map/values=] are [=ordered sets=] of {{ScopedCredentialDescriptor}}.
+
+    1. [=list/For each=] {{ScopedCredentialDescriptor}} |C| in |credentialDescriptorList|:
+        1. If <code>|C|.{{transports}}</code>
+            <dl class="switch">
+                : [=list/is not empty=]
+                :: [=list/For each=] value |T| of <code>|C|.{{transports}}</code>, [=set/append=] |C| to
+                    <code>|credentialsPerTransportMap|[|T|]</code>.
+
+                : [=list/is empty=]
+                :: [=list/For each=] value |T| of {{Transport}}, [=set/append=] |C| to
+                    <code>|credentialsPerTransportMap|[|T|]</code>.
+            </dl>
+
+    1. If |credentialsPerTransportMap|
+        <dl class="switch">
+            : [=map/size=] is not 0
+            :: the client SHOULD:
+                1. Let |transportHintsList| be the result of [=map/getting the keys=] of |credentialsPerTransportMap|.
+                1. Set |transport| to one of the values in |transportHintsList|. The client MAY use local configuration
```

Comment:
I think this means that an allowList of [{id: 1, transports: ["usb"]}, {id: 2, transports: ["nfc"]}] will call authenticatorGetAssertion on only one of credentials 1 or 2. I suspect that's not what was intended.